src/resolve/resolved-conf.c

```diff
@@ -565,6 +661,96 @@ static void read_proc_cmdline(Manager *m) {
                 log_warning_errno(r, "Failed to read kernel command line, ignoring: %m");
 }
 
+int config_parse_dnssec_algorithms_digests(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Manager *m = ASSERT_PTR(userdata);
+        int (*from_string)(const char*);
+        int (*init)(Manager*);
+        bool negative = false;
+        const char *object;
+        Set **set;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+
+        if (streq(lvalue, "DNSSECAlgorithms")) {
+                from_string = dnssec_algorithm_from_string;
+                init = manager_init_dnssec_algorithms;
+                set = &m->dnssec_algorithms;
+                object = "algorithm";
+        } else if (streq(lvalue, "DNSSECDigests")) {
+                from_string = dnssec_digest_from_string;
+                init = manager_init_dnssec_digests;
+                set = &m->dnssec_digests;
+                object = "digest";
+        } else
+                assert_not_reached();
+
+        r = set_ensure_allocated(set, NULL);
```

Comment:
as above.

note that the various set_xyz() calls all are fine with NULL pointers, and consider a NULL Set identical to an empty one